alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/action/GetDestinationsRequest.kt

```diff
@@ -29,19 +30,22 @@ class GetDestinationsRequest : ActionRequest {
     val srcContext: FetchSourceContext?
     val table: Table
     val destinationType: String
+    val filter: TermsQueryBuilder?
 
     constructor(
         destinationId: String?,
         version: Long,
         srcContext: FetchSourceContext?,
         table: Table,
-        destinationType: String
```

Comment:
Lets remove `destinationId` and `destinationType` since they are used for filtering the query. These query filters can be added to the attribute `filter` and we can change `filter` to be type: `List<TermsQueryBuilder>`